### PR TITLE
Engagement prediction & Basic GUI

### PR DESCRIPTION
**Pull request by IamMaxim**
Implement engagement prediction processor and make basic GUI that outputs I/O state plot, webcam preview and engagement plot.

### CODE REVIEW

**Review by illright — APPROVED** [review]

**Review by alkaitagi — APPROVED** [review]

File: vpt/cli/record.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        '''Stops the data stream.'''
+        self.running = False
+
+
+def record():
+    '''Runs all of the recorders to check that everything works correctly.'''
+    print('Recording...')
+
+    # Create capture nodes
+    engagement_source = Dummy()
```
Comment (aabounegm) [inline vpt/cli/record.py:43]: Shouldn't this be replaced by the actual estimator?

File: vpt/processors/video_engagement_estimator/__init__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from vpt.sources.base import SourceBase
+
+import numpy as np
+
+
+class VideoEngagementEstimator(ProcessorBase):
+    """Returns:
+        - none if no face was detected in the video frame
+        - true if face was detected and it is looking at the camera
+        - false if face was detected but it looks sideways
```
Comment (aabounegm) [inline vpt/processors/video_engagement_estimator/__init__.py:14]: So you didn't just use Boolean to represent a value with 2 states that is not related to truthiness, you use a nullable boolean to represent 3 states!
I did not want to comment on code quality, but come on..

**Review by aabounegm — COMMENTED** [review]
There are many possible code quality improvements (and pylint suggestions), but I guess we no longer care about that, right?

File: vpt/cli/record.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    keyboard_source = KeyboardSource()
+    mouse_source = MouseSource()
+
+    graph_display = GraphView(mouse_source, keyboard_source, engagement_source)
```
Comment (aabounegm) [inline vpt/cli/record.py:47]: The `GraphView` needs 4 parameters, the first of which is the video source

**Review by aabounegm — CHANGES_REQUESTED** [review]

File: vpt/processors/video_engagement_estimator/__init__.py (hunk 2 above)
Comment (IamMaxim) [reply vpt/processors/video_engagement_estimator/__init__.py:14]: This is how it is made in our state machine, that's not my decision

Comment (aabounegm) [reply vpt/processors/video_engagement_estimator/__init__.py:14]: I am referring to the use of the boolean variable, not the states. Like, use an `enum` or something

**Review by kuredoro — APPROVED** [review]
